source/agora/node/Runner.d

```diff
@@ -109,6 +116,30 @@ public Listeners runNode (Config config)
         result.flash = flash;
     }
 
+    TCPListener dnstcplistener;
+    scope (exit)
+        if (config.registry.enabled && config.registry.dns.enabled)
+        {
+            // Here, we might need to interrupt the task to correctly shut down,
+            // however this throws an exception in the task that needs to be explicitly
+            // handled. And it doesn't help with the error messages printed by Vibe.d.
+            //dnstask.interrupt();
+            dnstcplistener.stopListening();
+        }
```

Comment:
How did I miss this ? 🤦 
This closes the TCP listener as soon as this function exits. Instead, the listener should be stored in the `result`.

Reply:
Shoot